dashboard/src/actions/quisbyChartActions.js

```diff
@@ -0,0 +1,124 @@
+import * as TYPES from "./types.js";
+
+import { DANGER, ERROR_MSG } from "assets/constants/toastConstants";
+
+import API from "../utils/axiosInstance";
+import { showToast } from "./toastActions";
+import { uriTemplate } from "../utils/helper";
+
+const chartTitleMap = {
+  gb_sec: "Bandwidth",
+  trans_sec: "Transactions/second",
+  usec: "Latency",
+};
+
+export const getQuisbyData = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    dispatch({
+      type: TYPES.SET_ACTIVE_RESOURCEID,
+      payload: dataset.resource_id,
+    });
+    const response = await API.get(
+      uriTemplate(endpoints, "datasets_visualize", {
+        dataset: dataset.resource_id,
+      })
+    );
+    if (response.status === 200 && response.data.json_data) {
+      dispatch({
+        type: TYPES.GET_QUISBY_DATA,
+        payload: response.data.json_data,
+      });
+      dispatch({
+        type: TYPES.IS_UNSUPPORTED_TYPE,
+        payload: "",
+      });
+      dispatch(parseChartData());
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
+    if (error?.response) {
+      const errorMsg = error.response.data.message;
+      const isUnsupportedType = errorMsg?.toLowerCase().includes("unsupported");
+      if (isUnsupportedType) {
+        dispatch({
+          type: TYPES.IS_UNSUPPORTED_TYPE,
+          payload: errorMsg,
+        });
+      }
+    } else {
+      dispatch(showToast(DANGER, ERROR_MSG));
+    }
+
+    dispatch({ type: TYPES.COMPLETED });
+    dispatch({ type: TYPES.NETWORK_ERROR });
+  }
+};
+
+export const parseChartData = () => (dispatch, getState) => {
+  const response = getState().quisbyChart.data.data;
+  const COLORS = ["#8BC1F7", "#0066CC", "#519DE9", "#004B95", "#002F5D"];
+
+  const chartData = [];
+  const i = 0; // used to iterate over the colors, will be used for comparsion
```

Comment:
I don't think `i` is used the way you seem to think it is -- it wouldn't be much good for iteration, since it is `const` -- it's only used to index into `COLORS`, and I suspect that that doesn't do what we want it to (i.e, `i` is always zero at line 106 (unlike in line 105, which is a _different `i`_), so we're always using the same color!...what are the other ones for?).

Reply:
Other`COLORS` will be used for comparing more than one benchmark which is still under development.

Updated the code with single color